tests/virtualization/yast_virtualization.pm

```diff
@@ -49,6 +49,9 @@ sub run {
     x11_start_program('xterm');
     wait_screen_change { send_key 'alt-f10' };
     become_root;
+    zypper_call "se --provides 'libvirtd'";
```

Comment:
Why do you search libvirtd as key word and then install libvrit pkg here. 
Would you mind give us some details or comments? Thanks

 Besides, libvirt pkg just only index pkg, do not included any required files and tools and do not install by default. 
 If you want to install libvit tool stack, we'd better check and install libvirt-libs pkg. IMHO.

Reply:
The reason for searching "libvirtd" as a keyword and then installing the "libvirt" package is to ensure that the necessary libvirt components and services, including "libvirtd," are available on the system. While the "libvirt" package might indeed be an index package, it typically serves as a metapackage that depends on other essential packages and tools required for libvirt functionality.

In most cases, installing the "libvirt" package, which is a metapackage, automatically pulls in dependencies, such as "libvirt-libs," to provide a complete libvirt tool stack. This simplifies the installation process and ensures that all necessary components and tools are available.

Ultimately, the goal is to ensure that libvirt is correctly set up and that all required components and tools are available to support virtualization management effectively.

Reply:
Do you want me to change it to install `sudo zypper install libvirt-libs` and verify if it installs properly and verify the version and capabilities?

I can give it a try tomorrow, will be a learning :)

Reply:
Hi @varunkojha , the test failed because legacy libvirtd has been replaced with modular libvirt. So to make the test pass, we should not re-install an obsolete libvirtd, instead, we should adapt the test script to manipulate modular libvirt daemons(rather then libvirtd). Please read the link in the poo ticket discription again to get the background and find a solution finally.

https://progress.opensuse.org/issues/133862
> References:
> modular libvirt howtos: https://confluence.suse.com/pages/viewpage.action?pageId=1241481218

>Suggestions:
> reuse/extend/add modular libvirt lib functions in https://github.com/os-autoinst/os-autoinst-distri- opensuse/blob/master/lib/virt_autotest/utils.pm based on need, eg restart_modular_libvirt_daemons
> failure root cause seems to be in yast_virtualization in (re)starting libvirt in modular libvirt way

Reply:
Hi @varunkojha , the test failed because legacy libvirtd has been replaced with modular libvirt. So to make the test pass, we should not re-install an obsolete libvirtd, instead, we should adapt the test script to manipulate modular libvirt daemons(rather then libvirtd). Please read the link in the poo ticket discription to get the background and find a solution finally.

https://progress.opensuse.org/issues/133862
> References:
> modular libvirt howtos: https://confluence.suse.com/pages/viewpage.action?pageId=1241481218

>Suggestions:
> reuse/extend/add modular libvirt lib functions in https://github.com/os-autoinst/os-autoinst-distri- opensuse/blob/master/lib/virt_autotest/utils.pm based on need, eg restart_modular_libvirt_daemons
> failure root cause seems to be in yast_virtualization in (re)starting libvirt in modular libvirt way

Reply:
I tried working by making the changes to manipulate the modular libvirt daemons rather than installing the libvirtd.
Nothing seems working, it seems to me test clearly states that
libvirt is not installed but libvirt-libs is present and if I install libvirt-libs (says nothing to do already installed).
Also enable, start, and status of libvirtd.service we can't see because it is not present. 

Maybe there is some other daemon that I don't know needs to be addressed so that finally enables and starts libvirt and works.

Below is the list of test fails, and different approaches for extending and reusing the libvirt lib functions:
1. https://openqa.opensuse.org/tests/3658273#step/yast_virtualization/23
2. https://openqa.opensuse.org/tests/3658274#step/yast_virtualization/23
3. https://openqa.opensuse.org/tests/3658292#step/yast_virtualization/63
4. https://openqa.opensuse.org/tests/3659055#step/yast_virtualization/61
5. https://openqa.opensuse.org/tests/3659143#step/yast_virtualization/64
6. https://openqa.opensuse.org/tests/3659809#step/yast_virtualization/23

Reply:
You are quite near to the fix. https://openqa.opensuse.org/tests/3659143#step/yast_virtualization/63, you did correctly to start modular libvirt, but you should not start libvirtd.service at the end. libvirtd has been replaced with modular libvirt, libvirtd has not existed any more. There is not libvirt.service, libvirt is a toolkit or virtualization archeteture. when kvm/xen_tools are installed, libvirt(many related packages including libvirt-libs) will be installed. Some daemons are started by default but some daemons are required to be started manually, it varies on different OS. On TW, you are correct to start modular libvirt daemons manually because they are not enabled.

Reply:
Thanks Julie for putting it brief for me :)